book/forests.qmd

```diff
@@ -41,78 +57,65 @@ $$
+Deciding the number of minimum observations and/or the maximum depth can be performed with automated hyper-parameter optimisation.
+
+#### Terminal Node Predictions {.unnumbered .unlisted}
+
+The final major component of decision trees are *terminal node predictions*.
+As the name suggests, this is part of the algorithm that determines how to actually make a prediction for a new observation.
+A prediction is made for some new data by 'dropping' the new data 'down' the tree according to the optimal splits that were found during training.
+Returning to @fig-surv-ranfor, say a new data point is $x = \{`wt` = 3, `disp` = 250\}$, then in the first split the right branch is taken as `wt` $= 3 > 2.32$ and in the second split the left branch is taken as `disp` $= 250 \leq 258$, therefore the new data point 'lands' in the final leaf (the middle gray box).
+
+The resulting prediction is then a simple baseline statistic computed from the training data that fell into the corresponding node.
+In regression, this is most commonly (and simply) the sample mean of the training outcome data, but other terminal node predictions could be the sample median, a weighted average, or even a simple linear regression prediction from a model trained on the training data in the corresponding final leaf.
+In the example above, the final prediction would be $y = 20.755$, note how the final predictions are statistics based on training data, which means all potential predictions can be saved in the original trained mode and no complex computations are required during prediction.
+
+### Random Forests
+
+As is commonly the case in machine learning, higher interpretable yields poor predictive performance.
+Decision trees often overfit the training data, hence they have high variance, perform poorly on new data, and are not robust to even small changes in the original training data.
+Moreover, many important variables can be ignored as only a very small subset are selected, which can be particularly problematic in survival analysis when variables are often know *a priori* to have important clinical value.
+
+To counter these problems, *random forests* are preferred to improve prediction accuracy and decrease variance.
+Random forests utilise bootstrap aggregation, or *bagging*  [@Breiman1996a], to aggregate many decision trees.
+Bagging is a relatively simple algorithm, as follows:
+
+1. **For** $b = 1,...,B$:
+2. $D_b \gets \text{ Randomly sample with replacement } \dtrain$
+3. $\hatg_b \gets$ \text{ Train a decision tree on } $D_b$
+4. **end For**
+5. $g_B \gets \{\hatg_b\}^B_{b=1}$
+6. **return** $\hatg = g_B$
+
+Step $2$ is known as *bootstrapping*, which is the process of sampling a dataset, for example given data $\{1, 2, 3, 4\}$, standard sampling without replacement may result in the dataset $\{1, 3, 4\}$ as the first observation is drawn and removed from the original data, then the next two follow similarly.
```

Comment:
this is going too much into detail, I'd just say "step 2 is known as bootstraping, i.e. sampling with replacement."